src/pjpersist/datamanager.py

```diff
@@ -442,6 +447,11 @@ def __init__(self, conn, root_table=None):
         self._loaded_objects = {}
         self._inserted_objects = {}
         self._removed_objects = {}
+
+        self._readonly = None
```

Comment:
these are already set at the class level, no? I'd expec tthem to be in one place or the other (like dirty and pristine) but not both?

Reply:
It is, but we are calling `__init__` explicitly when transaction is completed, so we want to reset it to the original value. But I agree, I'll leave the class level with only type declarations.

Reply:
I know the `__init__()` pattern in common in Python and I introduced the usage in this package, but I think at this point a `reset()` method would be clearer that simply gets called from `__init__()`. All method uses of `__init__()` should be `reset()`.

Reply:
Introduced `_reset_data_manager()` to handle (re-)initialization to make code more conventional.